packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -972,10 +972,13 @@ class _DraggableScrollableSheetScrollPosition extends ScrollPositionWithSingleCo
     _ballisticControllers.add(ballisticController);
 
     double lastPosition = extent.currentPixels;
+    final bool wasExpanded = extent.isAtMax;
     void tick() {
       final double delta = ballisticController.value - lastPosition;
       lastPosition = ballisticController.value;
-      extent.addPixelDelta(delta, context.notificationContext!);
+      if (!wasExpanded || !extent.snap) {
+        extent.addPixelDelta(delta, context.notificationContext!);
```

Comment:
This feels like a bandaid. Theoretically `addPixelDelta` should not do anything if the gesture is in the upwards direction at the end of the extent, so if that's failing then the issue should be there. Also, it feels like this should not be blocked if going downwards.

Reply:
As in merge a disabled test? I don't think we've done that in the past. It would add dead code which is likely to cause confusion when future contributors are trying to gauge intent. 

I also don't have a ton of time at the moment to look into this issue, but it's very interesting so that's understandable. If you are unable to keep updating this PR, you can either close it and reopen it when you are able to have more time, or mark it as a draft if you want to keep going at it slowly. We typically will close PRs after a month of no response.

Reply:
Ok, thanks, will try to dig deeper

Reply:
Well, your suggestion helped, I think
Can you take a look at new fix?

Reply:
Closing issue as the code is reworked